app/scripts/components/leaflet/clustered-waterpoints.jsx

```diff
@@ -0,0 +1,66 @@
+import { Map } from 'leaflet';
+import { PruneCluster, PruneClusterForLeaflet } from './prune-cluster';
+import React, { PropTypes } from 'react';
+
+const ClusteredWaterpoints = React.createClass({
+  propTypes: {
+    data: PropTypes.array.isRequired,
+    map: PropTypes.instanceOf(Map).isRequired,
+  },
+  componentWillMount() {
+    this.pruneCluster = new PruneClusterForLeaflet();
+    this.pruneCluster.Cluster.Size = 21;
+    this.props.map.addLayer(this.pruneCluster);
+    this.markerIdMap = {};
+  },
+  componentDidMount() {
+    this.updateCluster();
+  },
+  shouldComponentUpdate(nextProps) {
+    return nextProps.data !== this.props.data;
+  },
+  componentDidUpdate() {
+    this.updateCluster();
+  },
+  componentWillUnmount() {
+    this.props.map.removeLayer(this.pruneCluster);
+    delete this.markerIdMap;
+  },
+  updateCluster() {
```

Comment:
Discussed [on slack](https://developmentgateway.slack.com/archives/tanzania-dashboards/p1444401267000247) -- I'll put some colours in